File: workspace/scala/src/test/scala/com/dslplatform/ocd/values/TestOneDoubleValue.scala
package com.dslplatform.ocd.values

import OneDoubleInValue.OneDoubleValue
import com.dslplatform.api.patterns.ServiceLocator
import org.scalatest._
import scala.reflect.runtime.universe._

class TestOneDoubleValue extends SpecExtensions {
  def `Test property field type: Double`() =
    checkType(
      typeOf[Double]
    , typeOf[OneDoubleValue].member("oneDouble": TermName).asMethod.returnType
    )

  def `Test property default value: 0.0`() {
    assert(OneDoubleValue().oneDouble === 0.0)
  }
}

File: workspace/java/src/test/java/com/dslplatform/ocd/values/TestOneDoubleValue.java
package com.dslplatform.ocd.values;

import com.dslplatform.client.Bootstrap;
import com.dslplatform.ocd.values.OneDoubleInValue.OneDoubleValue;
import com.dslplatform.patterns.ServiceLocator;
import org.junit.*;
import static org.junit.Assert.*;

public class TestOneDoubleValue {
    private static ServiceLocator locator;

    @BeforeClass
    public static void setUpClass() throws Exception {
//        locator = Bootstrap.init(TestOneDoubleValue.class.getResourceAsStream("dsl-project.ini"));
    }

    @AfterClass
    public static void tearDownClass() throws Exception {
//        locator.resolve(java.util.concurrent.ExecutorService.class).shutdown();
//        locator = null;
    }

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
    }

    @Test
    public void testFieldType() throws NoSuchFieldException {
        assertEquals(
                double.class,
                OneDoubleValue.class.getDeclaredField("oneDouble").getGenericType());
    }

    @Test
    public void testGetterType() throws NoSuchMethodException {
        assertEquals(
                double.class,
                OneDoubleValue.class.getMethod("getOneDouble").getGenericReturnType());
    }

    @Test
    public void testSetterType() throws NoSuchMethodException {
        assertEquals(
                OneDoubleValue.class,
                OneDoubleValue.class.getMethod("setOneDouble", double.class).getReturnType());
    }

    @Test
    public void testDefaultPropertyValue() {
        assertEquals(
                0.0,
                new OneDoubleValue().getOneDouble(),
                0.0);
    }
}
